app/lib/activitypub/activity/create.rb

```diff
@@ -4,10 +4,17 @@ class ActivityPub::Activity::Create < ActivityPub::Activity
   def perform
     return if delete_arrived_first?(object_uri) || unsupported_object_type?
 
-    status = find_existing_status
+    RedisLock.acquire(lock_options) do |lock|
+      status = find_existing_status
```

Comment:
Gotta do something like `if lock.acquired?`

Reply:
I think `status` must be first defined outside of the block, otherwise it's scoped to within the block only. Same thing in `process_status` below.